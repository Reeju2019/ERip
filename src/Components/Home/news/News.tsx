import React from 'react'
import Slider from 'react-slick'
import 'slick-carousel/slick/slick.css'
import 'slick-carousel/slick/slick-theme.css'
import newsData from '../../../Data/News.data.json'
import data from '../../../Data/const.data.json'
import { Container } from 'react-bootstrap'
import { Link } from 'react-router-dom'
import './news.css'

const News = () => {
  const settings = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 4,
    slidesToScroll: 1,
    responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 3,
          infinite: true,
          dots: true,
        },
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 2,
          initialSlide: 2,
        },
      },
      {
        breakpoint: 480,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
        },
      },
    ],
  }
  return (
    <>
      <section className='news-update'>
        <Container>
          {data.news.map((c: any, index) => {
            return (
              <div key={index}>
                <h4 className="news_heading">{c.heading}</h4>
                <Slider {...settings}>
                  {newsData.news.map((e: any, index) => {
                    return (
                      <div className="card device_card" key={index}>
                        <div className="img_card">
                          <img
                            src={e.news_image}
                            className="card-img-top"
                            alt="..."
                          />
                        </div>
                        <div className='card-body news_card_body'>
                          <h4 className='card-title news_title'>{e.des}</h4>
                          <Link to={''} className='news_more'>
                            {c.button}
                          </Link>
                        </div>
                      </div>
                    )
                  })}
                </Slider>
              </div>
            )
          })}
        </Container>
      </section>
    </>
  )
}

export default News
