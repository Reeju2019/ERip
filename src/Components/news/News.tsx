import React from "react";
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import { newsData } from "./newsData";
import { Container } from "react-bootstrap";
import { Link } from "react-router-dom";
import "./news.css";

const News = () => {
  const settings = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 4,
    slidesToScroll: 1,
    responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 3,
          infinite: true,
          dots: true,
        },
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 2,
          initialSlide: 2,
        },
      },
      {
        breakpoint: 480,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
        },
      },
    ],
  };
  return (
    <>
      <section className="news-update">
        <Container>
          <div>
            <h4 className="news_heading">News & Updates</h4>
            <Slider {...settings}>
              {newsData.map((e: any) => {
                return (
                  <div className="card device_card">
                    <div className="img_card">
                      <img
                        src={e.news_image}
                        className="card-img-top"
                        alt="..."
                      />
                    </div>
                    <div className="card-body news_card_body">
                      <h4 className="card-title news_title">{e.des}</h4>
                      <Link to={""} className="news_more">
                        Know More
                      </Link>
                    </div>
                  </div>
                );
              })}
            </Slider>
          </div>
        </Container>
      </section>
    </>
  );
};

export default News;
