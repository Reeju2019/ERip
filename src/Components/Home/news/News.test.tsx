import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import News from './News'
import { MemoryRouter as Router } from 'react-router-dom'

test('news', () => {
  render(
    <Router>
      <News />
    </Router>,
  )
})

describe('pnf rendering specification', () => {
  it('SnapShot testing', () => {
    const component = renderer.create(
      <Router>
        <News />
      </Router>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render without crashing', () => {
    const div = document.createElement('div')
    render(
      <Router>
        <News />
      </Router>,
    )
    ReactDOM.unmountComponentAtNode(div)
  })
})
