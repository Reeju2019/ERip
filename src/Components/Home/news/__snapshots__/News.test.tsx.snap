// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pnf rendering specification SnapShot testing 1`] = `
<section
  className="news-update"
>
  <div
    className="container"
  >
    <div>
      <h4
        className="news_heading"
      >
        News & Updates
      </h4>
      <div
        className="slick-slider slick-initialized"
        dir="ltr"
      >
        <div
          className="slick-list"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={null}
          onMouseMove={null}
          onMouseUp={[Function]}
          onTouchCancel={null}
          onTouchEnd={[Function]}
          onTouchMove={null}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="slick-track"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "WebkitTransform": "translate3d(0px, 0px, 0px)",
                "WebkitTransition": "",
                "msTransform": "translateX(0px)",
                "opacity": 1,
                "transform": "translate3d(0px, 0px, 0px)",
                "transition": "",
              }
            }
          >
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={-4}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/wa.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={-3}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/10.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={-2}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/media.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={-1}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/wa.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active slick-current"
              data-index={0}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/10.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active"
              data-index={1}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/media.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active"
              data-index={2}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/laptop.png"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active"
              data-index={3}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/wa.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={4}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/10.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={5}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/media.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={6}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/wa.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={7}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/10.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={8}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/media.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={9}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/laptop.png"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={10}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/wa.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={11}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/10.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={12}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/media.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={13}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card device_card"
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <div
                    className="img_card"
                  >
                    <img
                      alt="..."
                      className="card-img-top"
                      src="assets/news/wa.jpg"
                    />
                  </div>
                  <div
                    className="card-body news_card_body"
                  >
                    <h4
                      className="card-title news_title"
                    >
                      7 Reasons For Computer Lagging And How To Fix Them
                    </h4>
                    <a
                      className="news_more"
                      href="/"
                      onClick={[Function]}
                    >
                      Know More
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <ul
          className="slick-dots"
          onMouseEnter={null}
          onMouseLeave={null}
          onMouseOver={null}
          style={
            Object {
              "display": "block",
            }
          }
        >
          <li
            className="slick-active"
          >
            <button
              onClick={[Function]}
            >
              1
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              2
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              3
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              4
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              5
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              6
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              7
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
</section>
`;
